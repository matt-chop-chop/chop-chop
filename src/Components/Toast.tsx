import React from "react";
import { Flex, Image, Text } from "@chakra-ui/react";
import { CloseIcon } from "@chakra-ui/icons";

export interface SuccessToastProps {
  text: string;
  onClose?: () => void;
  background?: string;
}
const Toast = ({
  text,
  onClose,
  background = "var(--chakra-colors-light-success)",
}: SuccessToastProps) => {
  return (
    <Flex
      alignItems="center"
      background={background}
      border="none"
      borderRadius={4}
      justifyContent="space-between"
      pl={5}
      pr={2}
      py={2}
      width="260px"
    >
      <Text color="var(--chakra-colors-light-background)" variant="body">
        {text}
      </Text>
      {onClose && (
        <CloseIcon
          boxSize={3}
          color="var(--chakra-colors-light-background)"
          cursor="pointer"
          ml={5}
          onClick={onClose}
        />
      )}
    </Flex>
  );
};

export default Toast;
