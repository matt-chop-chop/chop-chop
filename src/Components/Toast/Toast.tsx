import React from "react";
import { Flex, Text } from "@chakra-ui/react";
import { CloseIcon } from "@chakra-ui/icons";

export type ToastProps = {
  text: string;
  onClose?: () => void;
  background?: string;
};

const Toast = ({
  text,
  onClose,
  background = "var(--chakra-colors-light-success)",
}: ToastProps) => {
  return (
    <Flex
      alignItems="center"
      background={background}
      border="none"
      borderRadius={4}
      justifyContent="space-between"
      px={5}
      py={2}
      width="260px"
    >
      <Text color="var(--chakra-colors-light-background)" variant="body">
        {text}
      </Text>
      {onClose && (
        <CloseIcon
          boxSize={3}
          color="var(--chakra-colors-light-background)"
          cursor="pointer"
          ml={5}
          onClick={onClose}
        />
      )}
    </Flex>
  );
};

export default Toast;
