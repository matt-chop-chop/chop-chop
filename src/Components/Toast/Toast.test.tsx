import { shallow } from "enzyme";
import React from "react";
import Toast, { ToastProps } from "./Toast";

const onClose = jest.fn();

const props: ToastProps = {
  text: "This is a test toast",
  onClose: onClose,
  background: "red",
};

describe("Toast", () => {
  it("is exported from Toast", () => {
    expect(Toast).toBeDefined();
  });

  it("renders as expected ", () => {
    const wrapper = shallow(<Toast {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("calls onClose when the close button is clicked ", () => {
    const wrapper = shallow(<Toast {...props} />);
    const CloseIcon = wrapper.find("CloseIcon");
    expect(CloseIcon).toHaveLength(1);
    const onClick = CloseIcon.prop("onClick") as () => void;
    onClick();
    expect(onClose).toHaveBeenCalledTimes(1);
  });

  it("renders with default props ", () => {
    const wrapper = shallow(<Toast text={props.text} />);

    expect(wrapper).toMatchSnapshot();
  });
});
