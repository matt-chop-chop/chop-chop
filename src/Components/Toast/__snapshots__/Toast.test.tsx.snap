// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast renders as expected  1`] = `
<Flex
  alignItems="center"
  background="red"
  border="none"
  borderRadius={4}
  justifyContent="space-between"
  px={5}
  py={2}
  width="260px"
>
  <Text
    color="var(--chakra-colors-light-background)"
    variant="body"
  >
    This is a test toast
  </Text>
  <CloseIcon
    boxSize={3}
    color="var(--chakra-colors-light-background)"
    cursor="pointer"
    ml={5}
    onClick={[MockFunction]}
  />
</Flex>
`;

exports[`Toast renders with default props  1`] = `
<Flex
  alignItems="center"
  background="var(--chakra-colors-light-success)"
  border="none"
  borderRadius={4}
  justifyContent="space-between"
  px={5}
  py={2}
  width="260px"
>
  <Text
    color="var(--chakra-colors-light-background)"
    variant="body"
  >
    This is a test toast
  </Text>
</Flex>
`;
